Application/EdFi.Ods.Api/Filters/EnforceAssignedProfileUsageFilter.cs

```diff
@@ -157,10 +157,11 @@ void SetForbiddenResponse()
                 $"Based on profile assignments, {(hasSingleProfile ? null : "one of ")}the following profile-specific content type{(hasSingleProfile ? null : "s")} is required when {(relevantContentTypeUsage == ContentTypeUsage.Readable ? "requesting" : "creating or updating")} this resource: '{string.Join("', '", assignedProfilesForRequest.OrderBy(a => a).Select(p => ProfilesContentTypeHelper.CreateContentType(resourceFullName.Name, p, relevantContentTypeUsage)))}'";
 
             var problemDetails = new SecurityAuthorizationException(
-                SecurityAuthorizationException.DefaultDetail,
+                SecurityAuthorizationException.DefaultDetail + " The request was not constructed correctly for the data policy that has been applied to this resource for the caller.",
                 errorMessage)
             {
-                CorrelationId = (string) _logContextAccessor.GetValue(CorrelationConstants.LogContextKey)
+                CorrelationId = (string) _logContextAccessor.GetValue(CorrelationConstants.LogContextKey),
+                InstanceTypeParts = ["data-policy", "incorrect-usage"]
```

Comment:
The requested type was: `security:data-policy:incorrect-usage`
But the change here results in: `security:authorization:data-policy:incorrect-usage`
Is this discrepancy acceptable?

Reply:
I seem to remember a conversation about this with Vinaya... and I think we decided that `data-policy` should not be under `authorization`, but directly under `security`. So, this is another good catch, and I think for this one we may need to introduce a `SecurityDataPolicyException`, with a relevant constructor for the usage.

@vimayya Can you confirm?

Reply:
@axelmarquezh, @gmcelhanon is this resulting in a 403 error or 400? Yes, I would avoid putting `data-policy` error under `authorization`.

Reply:
@vimayya It's returning a 403. 
I'll create a new exception (`SecurityDataPolicyException`) to accommodate the required type, we could change the http code if required. Should we go with http 400?

Reply:
@axelmarquezh Thanks. No change on the Http response status code at this time.

Reply:
@gmcelhanon @vimayya The security related errors now result in the next problem details:
```
{
  "detail": "A Data Policy failure was encountered. The request was not constructed correctly for the data policy that has been applied to this resource for the caller.",
  "type": "urn:ed-fi:api:security:data-policy:incorrect-usage",
  "title": "Data Policy Failure Due to Incorrect Usage",
  "status": 403,
  "correlationId": "0d8b59e778b1",
  "errors": [
    "Based on profile assignments, one of the following profile-specific content types is required when requesting this resource: 'application/vnd.ed-fi.school.test-profile-resource-child-collection-includeall.readable+json', 'application/vnd.ed-fi.school.test-profile-resource-includeall.readable+json'"
  ]
}
```

And for the bad request related errors I only changed the type to end with `data:data-policy`, so it results in:
```
{
  "detail": "The resource cannot be saved because a data policy has been applied to the request that prevents it.",
  "type": "urn:ed-fi:api:bad-request:data:data-policy",
  "title": "Data Policy Enforced",
  "status": 400,
  "correlationId": "ff125fb46f27",
  "errors": [
    "The Profile definition for 'test-profile-resource-non-creatable' excludes (or does not include) one or more required data elements needed to create the resource."
  ]
}
```

Does this match your expectations?